src/nwm_routing/src/nwm_routing/output.py

```diff
@@ -227,3 +234,7 @@ def nwm_output_generator(
             print("parity check complete")
         if showtiming:
             print("... in %s seconds." % (time.time() - start_time))
+            
+    if wrf_hydro_lastobs_file:
```

Comment:
we can and should write lastobs files even if a file isn't specified at first (i.e. if NOT wrf_hydro_lastobs_file).